### ISSUE TITLE
Provide a way to tag logging per request.

### ISSUE THREAD

**Issue by amol-** [TurboGears/tg2#89]
Currently when logging there is no way to know which request a log line belongs to, we might want to add a way to allow this. If this is not possible through standard logging module (which I suspect) we might provide (and use everywhere) a `tg.log` that logs the request id too when it's available.

**Comment by amol-** [comment]
`%(thread)s @ %(process)s` format might be a good enough solution for 90% of users. Should it be in the default quickstart?

**Comment by kiilerix** [comment]
Yeah, that would perhaps be helpful. Especially if the thread can be tracked all the way from the request comes in. Once you know it, it is quite trivial to add. But having it in the quickstart would remove the need for knowing it.

Something with ordering like `%(process)s - %(thread)s` would probably be more natural.
